src/AudioManager.ts

```diff
@@ -0,0 +1,158 @@
+      handleTranscript: options.handleTranscript,
+      handleError:
+        options.handleError ??
+        ((error: string) => console.error("Error:", error)),
+    };
+
+    this.audioInputsService = new AudioInputsService(
+      this.options.streamTimeslice,
+    );
+    this.audioInputsBrowser = new AudioInputsBrowser();
+    this.audioOutputsService = new AudioOutputsService();
+    this.mediaAudio = new MediaAudio();
+
+    // Listen to events from the AudioInputsService
+    this.audioInputsService.on("start", this.options.handleStartSTT);
+    this.audioInputsService.on("stop", this.options.handleStopSTT);
+    this.audioInputsService.on("transcript", this.options.handleTranscript);
+    this.audioInputsService.on("error", (error: string) =>
+      this.options.handleError?.(error),
+    );
+
+    // Listen to events from the AudioInputsBrowser
+    this.audioInputsBrowser.on("start", this.options.handleStartSTT);
+    this.audioInputsBrowser.on("stop", this.options.handleStopSTT);
+    this.audioInputsBrowser.on("transcript", this.options.handleTranscript);
+    this.audioInputsBrowser.on("error", (error: string) =>
+      this.options.handleError?.(error),
+    );
+  }
+
+  // **
+  // ** Audio Input ** //
+  // **
+  public startListening = (): void => {
+    if (this.options.sttService === "browser") {
+      this.audioInputsBrowser.startListening();
+    } else if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.startListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.options.duckVolumeLevel as number;
+    }
+  };
+
+  public stopListening = (): void => {
+    if (this.options.sttService === "browser") {
+      this.audioInputsBrowser.stopListening();
+    } else if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.stopListening();
+    }
+
+    if (this.mediaAudio.isPlaying) {
+      this.mediaAudio.volume = this.options.normalVolumeLevel as number;
+    }
+  };
+
+  public connect = (token: string): void => {
+    if (this.options.sttService === "charisma/deepgram") {
+      this.audioInputsService.connect(token);
+    }
+  };
+
+  public inputServiceResetTimeout = (timeout: number): void => {
+    this.audioInputsService.resetTimeout(timeout);
+  };
+
+  // **
+  // ** Browser STT Service ** //
+  // **
+  public browserIsSupported = (): boolean => {
+    return this.audioInputsBrowser.isSupported;
+  };
+
+  public browserResetTimeout = (timeout: number): void => {
+    this.audioInputsBrowser.resetTimeout(timeout);
+  };
+
+  // **
+  // ** Audio Outputs Service ** //
+  // **
+  public outputServicePlay = (
+    audio: ArrayBuffer,
+    options: boolean | AudioOutputsServicePlayOptions,
+  ): Promise<void> => {
+    return this.audioOutputsService.play(audio, options);
+  };
+
+  // **
+  // ** Media Audio ** //
+  // **
+  public mediaAudioPlay = (): Promise<void> => {
```

Comment:
Could this be simplified, because if every time we set mediaSrc we then play it we could have a simpler interface here?
And this will also need expanding to accommodate multiple audio sources passed through which optionally mute the previous (I think you already have a ticket for this?)

Reply:
I have a ticket for this. I'll add this note to that ticket though.